File: week2/day4/TheGardenApplication/TheGardenApplication/Garden.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TheGardenApplication
{
    class Garden
    {
        public List<Flower> Flowers { get; set; }
        public List<Tree> Trees { get; set; }
        public List<int> ThirstyFlowers { get; set; }
        public List<int> ThirstyTrees { get; set; }
        private int CountThirsty=0;
        public Garden()
        {
            this.Flowers=new List<Flower>();
            this.Trees=new List<Tree>();
            this.ThirstyFlowers=new List<int>();
            this.ThirstyTrees=new List<int>();
        }

        public void AddTree(Tree t)
        {
            Trees.Add(t);
        }

        public void AddFlower(Flower f)
        {
            Flowers.Add(f);
        }

        public void FindThirsty()
        {
            for (int i=0; i < Flowers.Count; i++)
            {
                
                if (Flowers[i].IfNeedWater())
                {
                    this.ThirstyFlowers.Add(i);
                    Console.WriteLine($"the {Flowers[i].Color} flower need water");

                    this.CountThirsty++;
                }
                else
                {
                    Console.WriteLine($"the {Flowers[i].Color} flower doesn't need water");
                }
            }
            for (int j = 0; j < Trees.Count; j++)
            {
              
                if (Trees[j].IfNeedWater())
                {
                    this.ThirstyTrees.Add(j);
                    Console.WriteLine($"the {Trees[j].Color} tree need water");
                    this.CountThirsty++;
                }
                else
                {
                    Console.WriteLine($"the {Trees[j].Color} flower doesn't need water");
                }
            }

        }

        public void watering(double water)
        {
            Console.WriteLine("Watering with " + water);
            this.FindThirsty();
            if (this.CountThirsty != 0)
            {
                double eachWater = water / this.CountThirsty;

                if (ThirstyTrees.Count != 0)
                {
                    foreach (int i in ThirstyTrees)
                    {
                        Trees[i].watering(eachWater);
                    }
                }

                if (ThirstyFlowers.Count != 0)
                {
                    foreach (int j in ThirstyFlowers)
                    {
                        Flowers[j].watering(eachWater);
                    }
                }
               
            }
            
        }

    }
}

File: DSA-2019/week-02/day-1/garden.py
from flower import Flower
from tree import Tree


class Garden():
    def __init__(self):
        self.trees = []
        self.flowers = []
        self.need_water_count = 0

    def add_tree(self, tree: Tree):
        self.trees.append(tree)

    def add_flower(self, flower: Flower):
        self.flowers.append(flower)

    def garden_info(self):
        info = ""
        for flower in self.flowers:
            if flower.need_water():
                info += f"the {flower.color} flower needs water\n"
            else:
                info += f"the {flower.color} flower doesn't' need water\n"
        for tree in self.trees:
            if tree.need_water():
                info += f"the {tree.color} flower needs water\n"
            else:
                info += f"the {tree.color} flower doesn't need' water\n"
        return info

    def count_need_water_planet(self):
        for flower in self.flowers:
            if flower.need_water():
                self.need_water_count += 1
        for tree in self.trees:
            if tree.need_water():
                self.need_water_count += 1

    def watering(self, water):
        self.count_need_water_planet()
        water_for_each = water / self.need_water_count
        for flower in self.flowers:
            if flower.need_water():
                flower.absorb_water(water_for_each)
        for tree in self.trees:
            tree.absorb_water(water_for_each)
        print(f"Watering with {water}\n" + self.garden_info())


tree1 = Tree("purple")
tree2 = Tree("orange")
flower1 = Flower("yellow")
flower2 = Flower("blue")
garden1 = Garden()
garden1.add_flower(flower1)
garden1.add_flower(flower2)
garden1.add_tree(tree1)
garden1.add_tree(tree2)
garden1.watering(40)
